### PR TITLE
Add some basic UI improvements for api-gateway services

### PR DESCRIPTION
**Pull request by andrewstucki**
### Description

There are a number of things we don't support yet, such as topologies, displaying upstreams, etc. for API gateways (the new VM-native ones), but I think these are the quickest things that we can add to clean up how our new gateways display in the UI. The fonts are slightly off due to running in proxying mode I guess?

#### Service Listing Page

Before:

Screen Shot 2023-02-28 at 9 57 39 PM

After:

Screen Shot 2023-02-28 at 9 56 42 PM

#### Search Bar

Before:

Screen Shot 2023-02-28 at 9 49 50 PM

After:

Screen Shot 2023-02-28 at 9 56 34 PM

#### Hovered Kind Label

Before:

Screen Shot 2023-02-28 at 9 50 11 PM

After:

Screen Shot 2023-02-28 at 9 56 04 PM

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: ui/packages/consul-ui/app/components/consul/kind/index.hbs
Comment (nathancoleman) [inline ui/packages/consul-ui/app/components/consul/kind/index.hbs:50]: Some of these seem like they're actually broken today. Would you mind fixing and verifying if you already have the setup?
```suggestion
          (array 'ingress-gateway' '/connect/gateways/ingress-gateway')
          (array 'terminating-gateway' '/connect/gateways/terminating-gateway')
          (array 'api-gateway' '/connect/gateways/api-gateway')
          (array 'mesh-gateway' '/connect/gateways/mesh-gateway')
```

File: ui/packages/consul-ui/app/components/consul/kind/index.hbs
Comment (nathancoleman) [inline ui/packages/consul-ui/app/components/consul/kind/index.hbs:37]: We only have [the Learn guide covering API Gateway for Kubernetes](https://developer.hashicorp.com/consul/tutorials/developer-mesh/kubernetes-api-gateway), so I suppose it makes sense to leave API gateway out of this list for now

**Review by nathancoleman — APPROVED** [review]
Nice! Some really high-value improvements without any complexity 👍 

Would like to see the currently-broken docs links fixed, but that's not an issue introduced by this PR

File: ui/packages/consul-ui/app/components/consul/kind/index.hbs
Comment (t-eckert) [reply ui/packages/consul-ui/app/components/consul/kind/index.hbs:50]: OOF, great catch Nate

File: ui/packages/consul-ui/app/components/consul/kind/index.hbs
Comment (andrewstucki) [reply ui/packages/consul-ui/app/components/consul/kind/index.hbs:37]: Yeah, I was thinking we could add it eventually. As is it'll still show a learn guide link, but it'll just link to the root of the learn guides, which I think is fine, didn't want to change this component around too much.

File: ui/packages/consul-ui/app/components/consul/kind/index.hbs
Comment (andrewstucki) [reply ui/packages/consul-ui/app/components/consul/kind/index.hbs:50]: So, it looks like we have redirects in place on our website, but the mesh one is still totally broken and 404s, so I'll just fix it and wait for a UI sign-off.

File: ui/packages/consul-ui/app/components/consul/kind/index.js
Comment (WenInCode) [inline ui/packages/consul-ui/app/components/consul/kind/index.js:18]: This works great as is, but if you'd like to update this component to use the newer ember (Octane) conventions for component you can update this to be:

```
import Component from '@glimmer/component';
import { titleize } from 'ember-cli-string-helpers/helpers/titleize';
import { humanize } from 'ember-cli-string-helpers/helpers/humanize';

const normalizedGatewayLabels = {
  'api-gateway': 'API Gateway',
  'mesh-gateway': 'Mesh Gateway',
  'ingress-gateway': 'Ingress Gateway',
  'terminating-gateway': 'Terminating Gateway',
};

export default class CounterComponent extends Component {
  get name() {
    const name = normalizedGatewayLabels[this.args.item.Kind];
    return name ? name : titleize(humanize(this.args.item.Kind));
  }
}
```

However, doing this will also require updating the associated template's usages of `item` and `withInfo` to be `@item` and `@withInfo` to denote the usage of passed in arguments.

Definitely not required, just providing it as an option.

https://guides.emberjs.com/release/components/component-arguments-and-html-attributes/#toc_arguments
and
https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/#component-properties__tracked-vs-cp

**Review by WenInCode — COMMENTED** [review]
For some reason the frontend tests on this are only running a `noop` job. Looking into it.

Changes look great though.

File: ui/packages/consul-ui/app/components/consul/kind/index.js
Comment (andrewstucki) [reply ui/packages/consul-ui/app/components/consul/kind/index.js:18]: Ooh, neat to know, that'd at least let us start removing some of the deprecation warnings that get thrown.

I'm thinking the bigger change can probably wait since we'll likely have some more changes going into this component in the next release i.e. at minimum, a learn tutorial -- and since we're trying to merge for a patch release, I don't want to make too many changes (especially not having touched Ember code for ~8 years).

File: ui/packages/consul-ui/tests/acceptance/dc/services/index.feature
Comment (WenInCode) [inline ui/packages/consul-ui/tests/acceptance/dc/services/index.feature:104]: I'm seeing this test fail locally still showing 2 service models. I can try and look into it to help out later if I have some time after some meetings

**Comment by WenInCode** [comment]
@andrewstucki any chance you could open this under a new branch name following the `ui/branch-name` branch name structure.

It looks like our [CI frontend tests only run if the branch has that naming structure.](https://github.com/hashicorp/consul/blob/b177dc4e2b2703cee260b77d37ac551544f12344/.circleci/config.yml#L1217-L1223)

**Comment by valeriia-ruban** [comment]
> @andrewstucki any chance you could open this under a new branch name following the `ui/branch-name` branch name structure.
> 
> It looks like our [CI frontend tests only run if the branch has that naming structure.](https://github.com/hashicorp/consul/blob/b177dc4e2b2703cee260b77d37ac551544f12344/.circleci/config.yml#L1217-L1223)

**Comment by andrewstucki** [comment]
Closing in favor of https://github.com/hashicorp/consul/pull/16508

**Pull request closed without merging**
